### ISSUE TITLE
Allow START WITH clause in ALTER SEQUENCE

### ISSUE THREAD

**Issue by dyemanov** [FirebirdSQL/firebird#6921]
Accordingly to the SQL spec, `ALTER SEQUENCE` includes either `RESTART WITH` clause (which changes the _NEXTVAL_) or _\<basic sequence generator option\>_ which means other sequence settings specified in the `CREATE SEQUENCE` statement. We allow to change _\<sequence generator increment\>_ in `ALTER SEQUENCE`, but don't allow to change _\<sequence generator start value\>_ which looks inconsistent and non-standard. Priorly, the _start value_ was changed by the `RESTART WITH` clause, but it was undone and now it cannot be altered at all.

**Comment by asfernandes** [comment]
In SQL 2013 it is:

```
<sequence generator definition> ::=
CREATE SEQUENCE <sequence generator name> [ <sequence generator options> ]

<sequence generator options> ::=
<sequence generator option>...

<sequence generator option> ::=
<sequence generator data type option>
| <common sequence generator options>

<common sequence generator options> ::=
<common sequence generator option>...

<common sequence generator option> ::=
<sequence generator start with option>
| <basic sequence generator option>

<basic sequence generator option> ::=
<sequence generator increment by option>
| <sequence generator maxvalue option>
| <sequence generator minvalue option>
| <sequence generator cycle option>
```

It does not allow to change START WITH, as it is not part of `basic sequence generator option`.

Is this changed in more recent versions?

**Comment by dyemanov** [comment]
You're correct re. SQL:2013, my oversight. However, PGSQL and MariaDB allow non-standard START WITH clause in ALTER SEQUENCE. And personally I'm on the same line, because dropping/recreating may be simply impossible due to dependencies. Oracle/MSSQL/DB2, however, are on the other side and strictly follow the standard.

**Comment by mrotteveel** [comment]
The problem with including a `START WITH` in `ALTER SEQUENCE` is what would be the semantics? Would it only update the stored start value, or would it also actually restart the sequence with that value? In addition, it creates an incompatibility with the current `CREATE OR ALTER SEQUENCE` behaviour (though maybe that is for the best).

**Comment by dyemanov** [comment]
@mrotteveel, I think it should only update the metadata (start value). RESTART may follow, if required.

**Comment by mrotteveel** [comment]
> @mrotteveel, I think it should only update the metadata (start value). RESTART may follow, if required.

I agree somewhat, but what about the behaviour of `CREATE OR ALTER SEQUENCE`? Currently `START WITH` in `CREATE OR ALTER` behaves as `ALTER SEQUENCE ... RESTART WITH ...` when the sequence already exists.
